src/s1reader/s1_orbit.py

```diff
@@ -341,3 +398,240 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
         warnings.warn(msg)
 
     return orbit_file_final
+
+
+def _is_orbitfile_cover_timeframe(orbit_file: str, t_start_stop_frame: list):
+    '''
+    Check if `orbitfile` covers `t_start_stop_frame`
+    Copied from `get_orbit_file_from_list()` and modified
+
+    Parameter
+    ---------
+    orbit_file: str
+
+    t_start_stop_frame: list(datetime.datetime)
+
+    Returns
+    -------
+    _: Bool
+        `True` if the orbit file covers the time range; False otherwise
+    '''
+
+    # get file name and extract state vector start/end time strings
+    t_orbit_start, t_orbit_stop = os.path.basename(orbit_file).split('_')[-2:]
+
+    # strip 'V' at start of start time string
+    t_orbit_start = datetime.datetime.strptime(t_orbit_start[1:], FMT)
+
+    # string '.EOF' from end of end time string
+    t_orbit_stop = datetime.datetime.strptime(t_orbit_stop[:-4], FMT)
+
+    # check if:
+    # 1. swath start and stop time > orbit file start time
+    # 2. swath start and stop time < orbit file stop time
+    return all(t_orbit_start < t < t_orbit_stop for t in t_start_stop_frame)
+
+
+def concatenate_resorb_from_list(zip_path: str, orbit_file_list: list) -> str:
```

Comment:
```suggestion
def concatenate_resorb_from_list(zip_path: str, orbit_file_list: list) -> str | None:
```
It looks like we're returning either string or `None` rather than an empty string